009-ui-and-admin-server-security.md

```diff
@@ -0,0 +1,107 @@
+The token can be a refresh token or an access token associated with the registered client which is obtained by the user using the [Strimzi cli](https://github.com/strimzi/strimzi-kafka-oauth/blob/master/examples/docker/kafka-oauth-strimzi/kafka/oauth.sh) tool.
+The admin server will extract the token from the header and configure the client properties before calling the broker using the standard Kafka OAUTHBEARER SASL mechanism.
+
+### UI using OAuth2 (OIDC)
+The OAuth2 flow for the UI is more complicated. The UI is responsible for obtaining the access token or refresh token to put onto the Authorization header on calls to the admin server. 
+ 
+An admin user will register a client with the OpenID Provider (OP) - to create an OIDC client that provides the `authorization_code` OAuth 2.0 flow. The UI server can then be configured to use the OpenID Provider (OP) for authentication, and generating access tokens.
+
+The UI server will use Passport.js to provide/handle an `authorization_code` interaction to authenticate user and retrieve an access token from the OP. Passport will also handle updating the session for the user to contain the access token, and refreshing/reauthentication when the token expires. 
+
+The access token can then be passed to admin API as a Bearer token Authorization header. Admin API can then use this to issue requests to Kafka via OAUTHBEARER (Kafka will authenticate against the OP).
+
+A OIDC compliant server is required over OAuth 2.0 for following reasons:
+1. It provides metadata at `<hostname>/.well-known/openid-configuration` - which can be used for service discovery by Passport, instead of requiring configuration of a series of endpoints.
+2. It can provide user profile information (if we eventually want to display that in the UI)
+
+![Authentication flow](./images/009-authentication-flow.svg)
+
+#### Example CR
+
+Much like the authentication section of `Kafka` oauth listeners, similar would be applied to the UI configuration. The difference being that a single 'dicoveryURI' is needed for discovering the OIDC endpoints.
+
+```
+backends:
+  - name: "cluster2"
+    authentication :
+      discoveryURI: "https://cluster2/auth/realms/{realm-name}"
+      clientSecret: 
+          name: "mysecret"
+          idKey: "clientId"
+          passwordKey: "clientSecret"
+      certificate:
+        name: "mycert"
+        certKey: "ca.p12"
+    adminApi:
+      uri: "https://admin-api"
+      certificate:
+        name: "mycert"
+        certKey: "ca.p12"
+
+``` 
+
+#### Proposed libraries
+- https://www.npmjs.com/package/passport - authentication middleware
+- https://www.npmjs.com/package/openid-client - Passport compatible OIDC client for communicating with an OP
+
+#### Outstanding questions
+1. Who is responsible for "authorization" checks that the UI will make to hide features from user. E.g - can user create a topic, can user access the UI, can user produce a message etc.
```

Comment:
`Admin.describeTopics()` can optionally include the authorized operations on the topic (see `org.apache.kafka.clients.admin.DescribeTopicsOptions#includeAuthorizedOperations(boolean)` and `org.apache.kafka.clients.admin.TopicDescription#authorizedOperations`) if that's what you mean?